models/datadictionary.go

```diff
@@ -0,0 +1,140 @@
+package models
+
+import (
+	"encoding/json"
+	"log"
+	"time"
+
+	"github.com/uc-cdis/cohort-middleware/config"
+	"github.com/uc-cdis/cohort-middleware/utils"
+)
+
+type DataDictionaryI interface {
+	GenerateDataDictionary() (*DataDictionaryModel, error)
+}
+
+type DataDictionary struct {
+	CohortDataModel CohortDataI
+}
+
+type DataDictionaryModel struct {
+	Total int64           `json:"total"`
+	Data  json.RawMessage `json:"data"`
+}
+
+type DataDictionaryEntry struct {
+	VocabularyID                     string          `json:"vocabularyID"`
+	ConceptID                        int64           `json:"conceptID"`
+	ConceptCode                      string          `json:"conceptCode"`
+	ConceptName                      string          `json:"conceptName"`
+	ConceptClassId                   string          `json:"conceptClassID"`
+	NumberOfPeopleWithVariable       int64           `json:"numberOfPeopleWithVariable"`
+	NumberOfPeopleWhereValueIsFilled int64           `json:"numberOfPeopleWhereValueIsFilled"`
+	NumberOfPeopleWhereValueIsNull   int64           `json:"numberOfPeopleWhereValueIsNull"`
+	ValueStoredAs                    string          `json:"valueStoredAs"`
+	MinValue                         float64         `json:"minValue"`
+	MaxValue                         float64         `json:"maxValue"`
+	MeanValue                        float64         `json:"meanValue"`
+	StandardDeviation                float64         `json:"standardDeviation"`
+	ValueSummary                     json.RawMessage `json:"valueSummary"`
+}
+
+var dataDictionaryResult *DataDictionaryModel = nil
+
+// Generate Data Dictionary Json
+func (u DataDictionary) GenerateDataDictionary() (*DataDictionaryModel, error) {
+
+	//Read from cache first
+	if dataDictionaryResult != nil {
+		return dataDictionaryResult, nil
+	}
+
```

Comment:
I think it would be good to break up the init / caching part into a separate method, so we have the following:
- at startup, main.go will call an initialization method, say "initDataDictionary"
- any calls to the endpoint would only look at the cached version. If it is not there yet, the endpoint would just return 503 (service unavailable) to indicate it is not ready yet.